### ISSUE TITLE
InputMap.action_has_event regression/bug (return false when it shouldn't)

### ISSUE THREAD

**Issue by EIREXE** [godotengine/godot#52733]
### Godot version

4317d3a4a26bb9ebaca7b788452f06f6688d8625

### System information

Windows 11 and Arch

### Issue description

InputMap.action_has_event is supposed to return true if an action has a matched event, recently it returns false when it used to return true.

I believe it might have been caused by 0e5c6e0d559f3206942608590067906fd793f37e

### Steps to reproduce

```
extends Node

func _ready():
	var actions = InputMap.get_action_list("test")
	print(InputMap.action_has_event("test", actions[0]))
```

### Minimal reproduction project

[InputMapBug.zip](https://github.com/godotengine/godot/files/7173990/InputMapBug.zip)

**Comment by EricEzaM** [comment]
Caused by the backport of the commit you noted: #50874

**Comment by akien-mga** [comment]
Fixed by #52737.
